### PR TITLE
Align observability test-utils with new service type changes.

### PR DESCRIPTION
**Pull request by nadundesilva**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/observability-test-utils/src/main/ballerina/Ballerina.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ name = "testobserve"
 version = "0.0.0"
 
 [[platform.java11.dependency]]
-path = "../libs/observability-test-utils-2.0.0-Preview7.jar"
+path = "../libs/observability-test-utils-2.0.0-Preview8-SNAPSHOT.jar"
```
Comment (rdhananjaya) [inline tests/observability-test-utils/src/main/ballerina/Ballerina.toml:7]: do we need to commit these toml changes

File: tests/observability-test-utils/src/main/java/org/ballerina/testobserve/listenerendpoint/Resource.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    public Resource(BObject serviceObject, ResourceFunctionType resourceFunctionType, String basePath) {
+        this.serviceObject = serviceObject;
+        this.resourceFunctionType = resourceFunctionType;
+        this.basePath = basePath;
+
+        // Validate resource
+        int paramCount = this.getParamTypes().length;
+        String resourcePath = this.getResourcePath();
+        if (paramCount > 2) {
```
Comment (rdhananjaya) [inline tests/observability-test-utils/src/main/java/org/ballerina/testobserve/listenerendpoint/Resource.java:46]: What does this validation do? Is this specific to the test case?

File: tests/observability-test-utils/src/main/ballerina/Ballerina.toml (hunk 1 above)
Comment (nadundesilva) [reply tests/observability-test-utils/src/main/ballerina/Ballerina.toml:7]: Most of the lang modules and other modules seem to have been committed with the new version. I can revert if this shouldn't be done.

File: tests/observability-test-utils/src/main/java/org/ballerina/testobserve/listenerendpoint/Resource.java (hunk 2 above)
Comment (nadundesilva) [reply tests/observability-test-utils/src/main/java/org/ballerina/testobserve/listenerendpoint/Resource.java:46]: This is a validation for the mock listener. The listener at the moment, only supports resource functions and only accepts two arguments.

The validation is there as a way to throw an error if a developer doesn't adhere to the expected format.

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**